### ISSUE TITLE
Newly Added Bookmark Not Displaying/Sorting Issue

### ISSUE THREAD

**Issue by murph146** [aaFn/Bookmark-search-plus-2#239]
Newly added bookmarks are not displaying in the sidebar. If you open the extension in a tab is does appear though.
The bookmark also comes up in the results even though it's not actually visible. 

I also don't see a way to sort the bookmarks in the Bookmarks Menu folder unless they are in a sub-folder.

**Comment by aaFn** [comment]
Hello @murph146 , any newly added bookmark displays immediately in the sidebar, when FF notifies BSP2.
Do you have a particular scenario you can describe where this is not happening ?

(for the sort menu action to be also accessible in the Bookmarks Menu folder, I will see to enable that)

**Comment by murph146** [comment]
Hello,

The issue happens under normal use. I click the ‘star’ to bookmark a new site and it does not show up on the Bookmark Search Plus 2 bar. If I search for the bookmark in BSP2 it shows as a result but when you click on it doesn’t take you to the location of the bookmark.

If need be I can do a remote session if you want to review further.

Sent from my iPhone

On May 8, 2022, at 7:22 PM, aaFn ***@***.***> wrote:

﻿

Hello @murph146<https://github.com/murph146> , any newly added bookmark displays immediately in the sidebar, when FF notifies BSP2.
Do you have a particular scenario you can describe where this is not happening ?

(for the sort menu action to be also accessible in the Bookmarks Menu folder, I will see to enable that)

**Comment by aaFn** [comment]
Hello @murph146 , that works for me like a charm.
In which location (= ²in which folder / path) do you try to save the bookmark, when you click on the 'star' ?

**Comment by murph146** [comment]
They are being saved to a folder called Mobile Bookmarks before I move them. I tried it on 3 different computers with the same results on all 3.

Does it only work when saving it to a default bookmarks folder? If need be I can create a screen recording showing the issue.

Thank you.

**Comment by Gitoffthelawn** [comment]
I'm just throwing this out as a possibility based on @murph146 mentioning that the folder is named `Mobile Bookmarks`.  Is there any chance this is some sort of sync issue or an issue where a folder only exists because it was created by sync?  Feel free to just answer with "nope" if you don't think so... no need to spend any time explaining why not unless you really want to. ;)

**Comment by aaFn** [comment]
Yes, that is the source of the problem. The Mobile Bookmarks folder doesn't systematically exists / show on all devices .. I am not sure of the logiic for it to be visible to add-ons.

I test for its existence at start of BSP2, and if there, then I display it, but if it doesn't exist, then I do not test for its existence again later, and nothing notifies the add-on that it is "appearing". So bookmarking to it, or to a folder inside it (if that is ever possible), doesn't show up later if the folder didn't already exist at start of the of BSP2 sesion.

**Comment by murph146** [comment]
I was hoping that would fix the issue but just saved a bookmark to the default bookmarks menu folder and while if I do a search for it after saving it says it’s there but it’s not visible in the sidebar.

Would you like me to do a screen recording? I can share the recording use OneDrive.

Sent from my iPhone

On May 11, 2022, at 3:02 AM, aaFn ***@***.***> wrote:

﻿

Yes, that is the source of the problem. The Mobile Bookmarks folder doesn't systematically exists / show on all devices .. I am not sure of the logiic for it to be visible to add-ons.

I test for its existence at start of BSP2, and if there, then I display it, but if it doesn't exist, then I do not test for its existence again later, and nothing notifies the add-on that it is "appearing". So bookmarking to it, or to a folder inside it (if that is ever possible), doesn't show up later if the folder didn't already exist at start of the of BSP2 sesion.

**Comment by aaFn** [comment]
See here my answer for the sort on the Bookmarks Manu folder -> https://github.com/aaFn/Bookmark-search-plus-2/issues/239#issuecomment-1120506587

For the save to Bookmarks Menu, that is working well on my side.
Maybe do a "Reload from FF API" in options to make sure you are properly synchronized, but there is no issue for that destination.

**Comment by murph146** [comment]
I messed with it some more today and I got the newly saved bookmarks to show up but only after collapsing and expanding the bookmarks menu folder where it’s saved which is kinda annoying but I guess it’s not the end of the world.

Did the screenshot come through showing the missing ‘sort bookmarks’ menu option?

Sent from my iPhone

On May 11, 2022, at 6:04 PM, aaFn ***@***.***> wrote:

﻿

See here my answer for the sort on the Bookmarks Manu folder -> #239 (comment)<https://github.com/aaFn/Bookmark-search-plus-2/issues/239#issuecomment-1120506587>

For the save to Bookmarks Menu, that is working well on my side.
Maybe do a "Reload from FF API" in options to make sure you are properly synchronized, but there is no issue for that destination.
